microk8s-resources/actions/enable.cilium.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+
+if ! which helm 2>&1 >/dev/null; then
+	echo "Cilium installation requires helm. Please install via 'snap install helm'." >&2
+	exit 1
+fi
+
+ARCH=$(arch)
+if ! [ "${ARCH}" = "amd64" ]; then
+  echo "Cilium is not available for ${ARCH}" >&2
+  exit 1
+fi
+
+echo "Enabling Cilium"
+
+if [ ! -f "${SNAP_DATA}/bin/cilium" ]
+then
+  CILIUM_VERSION="master"
+  CILIUM_ERSION=$(echo $CILIUM_VERSION | sed 's/v//g')
+  CILIUM_DIR="cilium-$CILIUM_ERSION"
+  SOURCE_URI="https://github.com/cilium/cilium/archive"
+  CILIUM_CNI_CONF="plugins/cilium-cni/05-cilium-cni.conf"
+  CILIUM_LABELS="k8s-app=cilium"
+  NAMESPACE=kube-system
+
+  echo "Fetching cilium version $CILIUM_VERSION."
+  sudo mkdir -p "${SNAP_DATA}/tmp/cilium"
+  (cd "${SNAP_DATA}/tmp/cilium"
+  sudo "${SNAP}/usr/bin/curl" -L $SOURCE_URI/$CILIUM_VERSION.tar.gz -o "$SNAP_DATA/tmp/cilium/cilium.tar.gz"
+  sudo gzip -f -d "$SNAP_DATA/tmp/cilium/cilium.tar.gz"
+  sudo tar -xf "$SNAP_DATA/tmp/cilium/cilium.tar" "$CILIUM_DIR/install" "$CILIUM_DIR/$CILIUM_CNI_CONF")
+
+  sudo mv "$SNAP_DATA/args/cni-network/cni.conf" "$SNAP_DATA/args/cni-network/10-kubenet.conf" 2>/dev/null || true
+  sudo cp "$SNAP_DATA/tmp/cilium/$CILIUM_DIR/$CILIUM_CNI_CONF" "$SNAP_DATA/args/cni-network/05-cilium-cni.conf"
+
+  (cd "${SNAP_DATA}/tmp/cilium/$CILIUM_DIR/install/kubernetes"
+  helm template cilium \
+      --namespace $NAMESPACE \
+      --set global.cni.confPath="$SNAP_DATA/args/cni-network" \
```

Comment:
Could we also set the `global.cni.binPath` so that the binary lands inside a directory owned by MicroK8s? We try not to pollute the host. Something like `$SNAP_DATA/opt/cni/bin` or `$SNAP_COMMON/opt/cni/bin` would be great.

Reply:
Is there a path that's writable? Cilium usually writes the CNI binary into `cni.binPath` in an init container to ensure it's installed and up to date. Alternatively I guess we could look at allowing Cilium to disable that option and manually install it into the right place in these scripts.

Reply:
Looking at just disabling the write into the CNI binary path: https://github.com/cilium/cilium/pull/8913

Reply:
Disabling the cni install looks great. I think that if we `--set global.cni.binPath="$SNAP_DATA/opt/cni/bin"` we will have similar results.